### PR TITLE
remove unhandledRejection process.exit()

### PR DESCRIPTION
**Pull request by BruceChar**
## Context

* Describe the changes made and why they were made.
 remove service exit strategy when unhandledRejection & unexpected occurs
* Provide links to any issues or designs for this change.

## Were any new external packages added?

* List external packages added and why they were added

### CODE REVIEW

**Review by zhihuizhangjaden — APPROVED** [review]

**Comment by crystalynliu** [comment]
I also see there is some `process.exit` in Redis and build script, could we remove it as well

**Comment by BruceChar** [comment]
>
the build script only run in build process. it means some command run failed or incompletely we need to check the service make sure that the build is total successfully.

**Comment by BruceChar** [comment]
if redis down, some api is unavailable.we need to catch the redis error and hold on

File: packages/app/src/index.ts [hunk 1]
```diff
@@ -28,7 +28,11 @@ app
   .use(accessControl)
   .use(router.routes())
 
-unexpectedListener()
+unexpectedListener((type: string, err: any) => {
+  if (type === 'unhandledRejection' && err.toString().includes('GraphQL Error')) {
```
Comment (xsteadybcgo) [inline packages/app/src/index.ts:32]: Why not keep the call without passing arguments and put the logic for the callback function in the definition of the unexpectedListener function? Adding arguments here seems unnecessary, you only need to call the unexpectedListener once at the root level.

**Review by xsteadybcgo — CHANGES_REQUESTED** [review]

File: packages/app/src/index.ts (hunk 1 above)
Comment (BruceChar) [reply packages/app/src/index.ts:32]: Just a expand reserve

**Comment by crystalynliu** [comment]
> if redis down, some api is unavailable.we need to catch the redis error and hold on

@BruceChar I think we talked it in the meeting, we have a alert to know it. we don't need to shut down the service. We shouldn't shut down api service anyway.

**Review by crystalynliu — APPROVED** [review]

**Review by xsteadybcgo — APPROVED** [review]

**Pull request merged**